manual/core/performance/README.md

```diff
@@ -0,0 +1,351 @@
+
+```java
+// No need to extract a CqlIdentifier, raw strings are handled efficiently:
+Row row = session.execute("SELECT * FROM user WHERE id = 1").one();
+row.getInt("age");
+
+PreparedStatement pst = session.prepare("UPDATE user SET name=:name WHERE id=:id");
+pst.bind().setInt("age", 25);
+```
+
+#### Type tokens
+
+[GenericType] is used to express complex generic types -- such as
+[nested collections](../#collection-types) -- in getters and setters. These objects are immutable
+and stateless, so they are good candidates for constants:
+
+```java
+public static final GenericType<Set<List<String>>> SET_OF_LIST_OF_STRING = new GenericType<Set<List<String>>>() {};
+
+Set<List<String>> teams = row.get("teams", SET_OF_LIST_OF_STRING);
+```
+
+`GenericType` itself already exposes a few of those constants. You can create your own utility class
+to store yours.
+
+#### Built queries
+
+Similarly, [built queries](../../query_builder/) are immutable and don't need a reference to a live
+driver instance. If you create them statically, they can be stored as constants:
+
+```java
+public static final BuildableQuery SELECT_SERVER_VERSION =
+    selectFrom("system", "local").column("release_version");
+```
+
+Note that you don't necessarily need to extract `CqlIdentifier` constants since the construction
+already happens at initialization time.
+
+#### Derived configuration profiles
+
+The configuration API allows you to build [derived profiles](../configuration/#derived-profiles) at
+runtime.
+
+```java
+DriverExecutionProfile dynamicProfile =
+  defaultProfile.withString(
+      DefaultDriverOption.REQUEST_CONSISTENCY, DefaultConsistencyLevel.EACH_QUORUM.name());
+```
+
+Their use is generally discouraged (you should define profiles statically in the configuration file
+as much as possible), but if there's no other way and you reuse them over time, store them instead
+of recreating them each time. 
+
+### Metadata
+
+The driver maintains [metadata](../metadata/) about the state of the Cassandra cluster. This work is
+done on dedicated "admin" threads (see the [thread pooling](#thread-pooling) section below), so it's
+not in direct competition with regular requests. 
+
+
+#### Filtering
+
+You can disable entire parts of the metadata with those configuration options: 
+
+```
+datastax-java-driver.advanced.metadata {
+  schema.enabled = true
+  token-map.enabled = true
+}
+```
+
+This will save CPU and memory resources, but you lose some driver features:
+
+* if schema is disabled, `session.getMetadata().getKeyspaces()` will always be empty: your
+  application won't be able to inspect the database schema dynamically.
+* if the token map is disabled, `session.getMetadata().getTokenMap()` will always be empty, and you
+  lose the ability to use [token-aware routing](../load_balancing/#token-aware).
+
+Note that disabling the schema implicitly disables the token map (because computing the token map
```

Comment:
TIL; so basically if you want token-aware routing you have to enable schema metadata entirely. Would it be worth exploring the idea of enabling token-aware routing (which is a feature that almost everyone wants) without fully enabling schema metadata (which is imo much less popular) – iow, only fetch the relevant schema bits that are required to compute the token map?

Reply:
Yes, the only thing the token map needs are keyspace replication options.

That being said, getting rid of partial schema updates in driver 4 was a tremendous simplification of the internal code, quite honestly I feel reluctant to reintroduce some of it.

I suggest we wait. I don't see any way this could break the API, so If users ask for it, we can address it later.